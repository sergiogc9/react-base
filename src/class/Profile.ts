export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
