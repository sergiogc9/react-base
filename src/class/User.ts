export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
