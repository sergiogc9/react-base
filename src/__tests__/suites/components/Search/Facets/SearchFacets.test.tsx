import React from "react";
import { shallow } from "enzyme";
import { RouterState } from 'connected-react-router';

import SearchFacets from "@src/components/Search/Facets/SearchFacets";
import { FacetItemsGroup, FacetObject, Facet } from "@src/class/Facet";

const facetsGroups: FacetItemsGroup = {
	'channel_type_id': [{ key: "24", counter: 22 }],
	'country_path': [{ key: "0534", counter: 100 }, { key: "0103", counter: 50 }],
	'language_id': [{ key: "127", counter: 250 }],
	'language.id': [{ key: "127", counter: 250 }] // TODO (GIS-2327): Remove it once we have all with the same field
};

const facets: FacetObject = {
	groups: facetsGroups
};

const routerLocation: RouterState["location"] = {
	pathname: "/article",
	search: "",
	state: "",
	hash: ""
};

let onFetchFocusList: any;

describe("<SearchFacets />", () => {
	let wrapper: any;

	beforeEach(() => {
		onFetchFocusList = jest.fn();
		wrapper = shallow(<SearchFacets
			location={routerLocation}
			facets={facets}
			focusList={[]}
			loadingFacets={false}
			onFetchFocusList={onFetchFocusList}
			showMoreFacetGroupKey={null}
		/>);
	});

	it("Test component snapshot", () => {
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onFetchFocusList).not.toHaveBeenCalled();
	});

	it("Test component snapshot without focus list with article behaviour", () => {
		wrapper = shallow(<SearchFacets
			location={routerLocation}
			facets={facets}
			focusList={null}
			loadingFacets={false}
			onFetchFocusList={onFetchFocusList}
			showMoreFacetGroupKey={null}
		/>);
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onFetchFocusList).toHaveBeenCalled();
	});

	it("Test component snapshot without focus list with definition behaviour", () => {
		wrapper = shallow(<SearchFacets
			location={{ ...routerLocation, pathname: "/focus/id/feed/id" }}
			facets={facets}
			focusList={null}
			loadingFacets={false}
			onFetchFocusList={onFetchFocusList}
			showMoreFacetGroupKey={null}
		/>);
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onFetchFocusList).not.toHaveBeenCalled();
	});

	it("Test component snapshot with show more facet group key", () => {
		wrapper.setProps({ showMoreFacetGroupKey: Facet.channelGroupKey });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Loading component should not be displayed if previous facets are loaded", () => {
		wrapper.setProps({ loadingFacets: true, facets: null });
		expect(wrapper.find("CircularProgress").exists()).toEqual(true);
		wrapper.setProps({ loadingFacets: true, facets });
		expect(wrapper.find("CircularProgress").exists()).toEqual(false);
	});

	it("Test with empty facets", () => {
		wrapper.setProps({ focusList: [], facets: null});
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
