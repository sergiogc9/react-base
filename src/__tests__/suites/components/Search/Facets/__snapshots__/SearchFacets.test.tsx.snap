// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFacets /> Test component snapshot 1`] = `
"<Fragment>
  <div id=\\"searchFacetsGroups\\">
    <Connect(SearchFacetsFeedsGroup) />
    <Connect(SearchFacetsGroup) groupKey=\\"tenants.categories_id\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"tenants.tags\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"channel_type_id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"media_id\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"language_id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"country_path\\" groupItems={{...}} />
  </div>
</Fragment>"
`;

exports[`<SearchFacets /> Test component snapshot with show more facet group key 1`] = `
"<Fragment>
  <div id=\\"searchFacetsGroups\\">
    <Connect(SearchFacetsFeedsGroup) />
    <Connect(SearchFacetsGroup) groupKey=\\"tenants.categories_id\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"tenants.tags\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"channel_type_id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"media_id\\" groupItems={[undefined]} />
    <Connect(SearchFacetsGroup) groupKey=\\"language_id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"country_path\\" groupItems={{...}} />
  </div>
  <Connect(SearchFacetsDialog) groupKey=\\"channel_type_id\\" />
</Fragment>"
`;

exports[`<SearchFacets /> Test component snapshot without focus list with article behaviour 1`] = `
"<Fragment>
  <div id=\\"searchFacetsGroups\\">
    <CircularProgress id=\\"searchFacetsGroupsLoading\\" style={{...}} scale={1} determinateDashoffset={187} centered={true} />
  </div>
</Fragment>"
`;

exports[`<SearchFacets /> Test component snapshot without focus list with definition behaviour 1`] = `
"<Fragment>
  <div id=\\"searchFacetsGroups\\">
    <Connect(SearchFacetsGroup) groupKey=\\"channel_type_id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"language.id\\" groupItems={{...}} />
    <Connect(SearchFacetsGroup) groupKey=\\"country_path\\" groupItems={{...}} />
  </div>
</Fragment>"
`;

exports[`<SearchFacets /> Test with empty facets 1`] = `
"<Fragment>
  <div id=\\"searchFacetsGroups\\">
    <Connect(SearchFacetsFeedsGroup) />
  </div>
</Fragment>"
`;
